src/snmalloc/ds/flaglock.h

```diff
@@ -56,24 +59,19 @@ namespace snmalloc
     }
 
   private:
-    using ThreadIdentity = int const*;
-
     /**
      * @brief owner
-     * We use a pointer to TLS field as the thread identity.
-     * std::thread::id can be another solution but it does not
-     * support `constexpr` initialisation on some platforms.
+     * We use the Pal to provide the ThreadIdentity.
      */
-    std::atomic<ThreadIdentity> owner = nullptr;
+    std::atomic<ThreadIdentity> owner = ThreadIdentity();
 
     /**
      * @brief get_thread_identity
      * @return The identity of current thread.
      */
-    inline ThreadIdentity get_thread_identity()
+    static size_t get_thread_identity()
```

Comment:
I think this should still be a `ThreadIdentity` even if they're always `size_t`-s right now.